robottelo/hosts.py

```diff
@@ -479,6 +479,118 @@ def install_cockpit(self):
         if result.status != 0:
             raise ContentHostError('Failed to install the cockpit')
 
+    def register(
+        self,
+        target=settings.server.hostname,
+        org='Default_Organization',
+        loc='Default_Location',
+        activation_keys=None,
+        setup_insights=False,
+        setup_remote_execution=True,
```

Comment:
There is already a `setup_rex` function on the ContentHost class, why have this function also control REX setup?

Reply:
setup_rex is legacy, though I think both have valid use case. You may want to enable rex straight when registering host. Or you already have a registered non-rex host and you just want to enable it afterwards